src/reporter.test.ts

```diff
@@ -0,0 +1,47 @@
+import { z } from "zod";
+import { parseEnv, port } from "./compat.js";
+
+describe("reporter", () => {
+  it("should test the error output", () => {
+    expect(() =>
+      parseEnv(
+        {
+          FOO: "bar",
+          CONFIG_OBJECT: JSON.stringify({ a: 123, c: { d: 21 } }),
+          DO_NOT_USE_THIS_KEY: "this value might not exist",
+        },
+        {
+          CONFIG_OBJECT: z
+            .object({
+              a: z.string(),
+              b: z.number(),
+              c: z.object({
+                d: z.string(),
+              }),
+            })
+            .meta({
+              description:
+                "This is a very special object that holds configuration.",
+            }),
+          BLA: z.string().default("bar"),
+          FOO: z.number(),
+          DO_NOT_USE_THIS_KEY: z.string().meta({
+            deprecated: true,
```

Comment:
I think this is a great idea! Though checking `deprecated` in metadata is not actually implemented here unless I'm missing something, right?